miniserver/routes/middleware.js

```diff
@@ -0,0 +1,120 @@
+  try {
+    let command
+    const requestor = await User.findByPhone({ phone: senderPhoneNumber })
+    if (smsParams.length < 1) {
+      response.message('error: empy sms command')
+      return res.send(response.toString())
+    }
+    if (smsParams[0] === 'b') {
+      command = 'balance'
+      req.processedBody = { ...req.processedBody, command, requestor }
+    } else if (smsParams[0] === 'p') {
+      command = 'pay'
+      let funder
+      let amount
+      if (smsParams.length < 2) {
+        response.message('error: pay request requires funder and an amount. example request "p +14158401999 0.1"')
+        return res.send(response.toString())
+      }
+      // Allow requesting of funds from users by address (without checking registered phone number)
+      if (smsParams[1].substr(0, 2) === '0x') {
+        if (ethers.utils.isAddress(smsParams[1])) {
+          funder.address = smsParams[1]
+        } else {
+          response.message(`error: invalid funder address ${smsParams[1]}. example request "p 0x8ba1f109551bd432803012645ac136ddd64dba72 0.1"`)
+          return res.send(response.toString())
+        }
+      } else {
+        // set country and look up user address from Phone Number
+        let funderPhone = smsParams[1]
+        if (funderPhone.substr(0, 1) !== '+') {
+          const { countryIso3 } = phone(requestor.phone)
+          const { isValid, phoneNumber } = phone(funderPhone, countryIso3)
+          funderPhone = phoneNumber
+          if (!isValid) {
+            response.message(`error: invalid recipient phone number ${smsParams[1]}. example request "p +14158401999 0.1"`)
+            return res.send(response.toString())
+          }
+        }
+        Logger.log(`funderPhone: ${funderPhone}`)
+        funder = await User.findByPhone({ phone: funderPhone })
+        if (!funder?.address) {
+          response.message(`error: funders phone number is not a registered wallet: ${smsParams[1]}. example request "p +14158401999 0.1"`)
+          return res.send(response.toString())
+        }
+      }
+
+      if (toNumber(smsParams[2]) > 0) {
+        amount = ethers.utils.parseEther(smsParams[2])
+      } else {
+        response.message(`error: pay request requires a valid amount': ${smsParams[2]} example request "p +14158401999 0.1"`)
+        return res.send(response.toString())
+      }
+      req.processedBody = { ...req.processedBody, command, requestor, funder, amount }
+    } else {
+      response.message('error: invalid sms command. example payment request "p +14158401999 0.1"')
+      return res.send(response.toString())
+    }
+  } catch (ex) {
+    console.error(ex)
+    const { code, error, success } = parseError(ex)
+    response.message(`An exception occured. success: ${success} code: ${code} error: ${error.substr(0, 100)} `)
+    return res.send(response.toString())
+  }
+  next()
+}
+
+const validateID = async (req, res, next) => {
+  const { id: unvalidatedID } = req.body
```

Comment:
I think it is best to have phone and address as separate parameters, with fixed type

Reply:
Have changed this now.
Note: The reason it used one field was for the use case similar to when requesting funds from the operator where we could dynamically determine whether a phone number or address was parsed.
Having said that, if developers are calling this API then more structure makes sense.
Moving forward we should evaluate this vs `checkExistence` in the server as they have similar functionality.